core/src/main/java/com/yahoo/imapnio/async/request/UidFetchCommand.java

```diff
@@ -46,7 +84,19 @@ public UidFetchCommand(@Nonnull final String uids, @Nonnull final String items)
      * @param macro the macro, for example, ALL
      */
     public UidFetchCommand(@Nonnull final String uids, @Nonnull final FetchMacro macro) {
-        super(true, uids, macro);
+        this(uids, macro, null);
+    }
+
+    /**
+     * Initializes a {@link UidFetchCommand} with string form uids and macro.
+     *
+     * @param uids the UID string following the RFC3501 syntax. For ex:3857529045,3857529047:3857529065
+     * @param macro the macro, for example, ALL
+     * @param partialExtUidFetchInfo partial extension uid fetch info
+     */
+    public UidFetchCommand(@Nonnull final String uids, @Nonnull final FetchMacro macro,
```

Comment:
not critical but a pattern. Let's not let user to pass in @Nullable partialExtUidFetchInfo, a nullable value makes parameter list looks confusing (required or not required? such). Change this to @Nonnull instead, and let line 87 to call super(...)